src/ReorderEligibility/ItemsOutOfStockEligibilityChecker.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\CustomerReorderPlugin\ReorderEligibility;
+
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+
+final class ItemsOutOfStockEligibilityChecker implements ReorderEligibilityChecker
+{
+    /** @var ReorderEligibilityConstraintMessageFormatterInterface */
+    private $reorderEligibilityConstraintMessageFormatter;
+
+    public function __construct(
+        ReorderEligibilityConstraintMessageFormatterInterface $reorderEligibilityConstraintMessageFormatter
+    ) {
+        $this->reorderEligibilityConstraintMessageFormatter = $reorderEligibilityConstraintMessageFormatter;
+    }
+
+    public function check(OrderInterface $order, OrderInterface $reorder): array
+    {
+        $variantsOutOfStock = [];
+
+        /** @var OrderItemInterface $orderItem */
+        foreach ($order->getItems()->getValues() as $orderItem) {
+            /** @var ProductVariantInterface $productVariant */
+            $productVariant = $orderItem->getVariant();
+            if ($productVariant->isTracked() && !$productVariant->isInStock()) {
+                array_push($variantsOutOfStock, $orderItem->getVariantName());
+            }
+        }
+
+        if (empty($variantsOutOfStock)) {
+            return [];
+        }
+
+        return [
+            'type' => 'info',
+            'message' => 'sylius.reorder.items_out_of_stock',
+            'parameters' => [
+                '%order_items%' => $this->reorderEligibilityConstraintMessageFormatter->format($variantsOutOfStock),
+            ]
```

Comment:
I'm wondering does ReorderEligibility**Checker** should return a serialized flash message... It's against SRP principle for me. What I would expect from such a service is returning boolean flag or event better - throwing an exception if something is wrong. Then these exceptions can be caught where the main checker is used and passed to some "notifier" that would create a flash message basing on an exception.